File: backend/app/routes/enrollments.py
from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity

from app.models.enrollment import create_enrollment, get_user_enrollments, serialize_enrollment
from app.models.course import get_course_by_id, serialize_course
from app.utils.validators import validate_required_fields, validate_email

enrollments_bp = Blueprint('enrollments', __name__, url_prefix='/api/enrollments')


# -------------------------------------------------------------------
# POST /api/enrollments
# -------------------------------------------------------------------
@enrollments_bp.route('', methods=['POST'])
def enroll():
    data = request.get_json(silent=True) or {}

    missing = validate_required_fields(data, ['course_id', 'full_name', 'email'])
    if missing:
        return jsonify({"success": False, "message": f"Missing required fields: {', '.join(missing)}"}), 400

    if not validate_email(data['email']):
        return jsonify({"success": False, "message": "Invalid email format"}), 400

    # Verify the course exists
    course = get_course_by_id(data['course_id'])
    if not course:
        return jsonify({"success": False, "message": "Course not found"}), 404

    enrollment_id = create_enrollment({
        'user_id': data.get('user_id', ''),
        'course_id': data['course_id'],
        'full_name': data['full_name'].strip(),
        'email': data['email'],
        'phone': data.get('phone', ''),
    })

    return jsonify({
        "success": True,
        "data": {"enrollment_id": enrollment_id},
        "message": "Enrollment successful",
    }), 201


# -------------------------------------------------------------------
# GET /api/enrollments/my  (protected)
# -------------------------------------------------------------------
@enrollments_bp.route('/my', methods=['GET'])
@jwt_required()
def my_enrollments():
    user_id = get_jwt_identity()
    enrollments = get_user_enrollments(user_id)

    result = []
    for enrollment in enrollments:
        item = serialize_enrollment(enrollment)
        # Attach course info
        course = get_course_by_id(enrollment.get('course_id', ''))
        item['course'] = serialize_course(course)
        result.append(item)

    return jsonify({
        "success": True,
        "data": result,
    }), 200

File: backend/routes/enrollments.js
import express from 'express';
import { getDB, ObjectId } from '../db.js';
import { authMiddleware } from '../middleware/auth.js';

const router = express.Router();

// POST /api/enrollments
router.post('/', async (req, res) => {
  try {
    const { course_id, full_name, email, phone, user_id } = req.body || {};
    if (!course_id || !full_name || !email) {
      return res.status(400).json({ success: false, message: 'Missing required fields: course_id, full_name, email' });
    }

    const db = getDB();
    let course;
    try {
      course = await db.collection('courses').findOne({ _id: new ObjectId(course_id) });
    } catch {
      course = await db.collection('courses').findOne({ slug: course_id });
    }

    if (!course) {
      return res.status(404).json({ success: false, message: 'Course not found' });
    }

    const enrollmentDoc = {
      user_id: user_id || '',
      course_id: course._id.toString(),
      full_name: full_name.trim(),
      email: email.toLowerCase().trim(),
      phone: phone || '',
      created_at: new Date(),
    };

    const result = await db.collection('enrollments').insertOne(enrollmentDoc);

    return res.status(201).json({
      success: true,
      data: { enrollment_id: result.insertedId.toString() },
      message: 'Enrollment successful',
    });
  } catch (err) {
    return res.status(500).json({ success: false, message: 'Server error' });
  }
});

// GET /api/enrollments/my
router.get('/my', authMiddleware, async (req, res) => {
  try {
    const db = getDB();
    const enrollments = await db.collection('enrollments').find({ user_id: req.userId }).toArray();

    const result = [];
    for (const e of enrollments) {
      let course = null;
      if (e.course_id) {
        try {
          course = await db.collection('courses').findOne({ _id: new ObjectId(e.course_id) });
        } catch {}
      }
      result.push({
        id: e._id.toString(),
        course_id: e.course_id,
        full_name: e.full_name,
        email: e.email,
        phone: e.phone,
        course: course ? {
          id: course._id.toString(),
          title: course.title,
          slug: course.slug,
          image: course.image,
          level: course.level,
        } : null,
      });
    }

    return res.json({ success: true, data: result });
  } catch (err) {
    return res.status(500).json({ success: false, message: 'Server error' });
  }
});

export default router;
